### ISSUE TITLE
pyinstallation.dependency() version check broken?

### ISSUE THREAD

**Issue by infirit** [mesonbuild/meson#6792]
**Describe the bug**
``.dependency()`` on the python module does not check version

**To Reproduce**
``
pymod = import('python')
pyinstall = pymod.find_installation()
pyinstall.dependency(version: '>=3.20', required: true)

**Expected behavior**
Clearly there is no python version 3.20 so this should fail.

**system parameters**
Native gentoo linux build
Happens with any python3 version
meson version 0.53.1
ninja version 1.9.0.git.kitware.dyndep-1.jobserver-1

I need to check for a specific minimal python version. Either I am missing something or this check is broken.

The version check works fine with a "normal" ``dependency()``

I tried mesonbuild channel on freenode 3 times but although someone told me it is not dead it is pretty dead when there is no action for hours after asking,

**Comment by infirit** [comment]
A bit odd to mark this issue, which is mentioned in commit 044c1673b6963f2e610633eed6401139b6fa3848, as duplicate 5 years after it was fixed 😕

**Comment by bonzini** [comment]
Yes, should have marked it the other way round. On the ohter hand for some reason this one was still open and #9504 was closed...
